includes/Core/McpClient.php

```diff
@@ -0,0 +1,541 @@
+<?php
+/**
+ * MCP Client for connecting to external MCP servers.
+ *
+ * @package McpAdapter
+ */
+
+declare( strict_types=1 );
+
+namespace WP\MCP\Core;
+
+use WP\MCP\Infrastructure\ErrorHandling\Contracts\McpErrorHandlerInterface;
+use WP\MCP\Infrastructure\Observability\Contracts\McpObservabilityHandlerInterface;
+
+/**
+ * MCP Client for connecting to external MCP servers.
+ * 
+ * Mirrors the McpServer architecture for consistency.
+ */
+class McpClient {
```

Comment:
Buys us time to shape the API. There's currently no way to overload this class anyway, and there's a significant number of public methods on this class.

```suggestion
final class McpClient {
```